torchtext/experimental/vocab.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from typing import Dict, List
+
+import torch
+import torch.nn as nn
+
+logger = logging.getLogger(__name__)
+
+
+class Vocab(nn.Module):
+    r"""Creates a vocab object which maps tokens to indices.
+
+    Note that the ordering in which key value pairs were inserted in the `ordered_dict` will be respected when building the vocab.
+    Therefore if sorting by token frequency is important to the user, the `ordered_dict` should be created in a way to reflect this.
+
+    Arguments:
+        ordered_dict (collections.OrderedDict): object holding the frequencies of each token found in the data.
+        min_freq: The minimum frequency needed to include a token in the vocabulary.
+            Values less than 1 will be set to 1. Default: 1.
+        specials: The tuple of special tokens (e.g., padding or eos) that will be prepended/postpended to the vocabulary.
+            based on the `specials_first` flag. The ordering of the tuple will be preserved. Default: ('<unk>', '<pad>')
+        specials_first: Whether to add special tokens into the vocabulary at first. If it is False,
+            they are added into the vocabulary at last. Default: True.
+
+    Raises:
+        ValueError: if a default `unk_token` isn't provided.
+    """
+
+    def __init__(self, ordered_dict, min_freq=1, unk_token='<unk>', specials=('<unk>', '<pad>'), specials_first=True):
+        super(Vocab, self).__init__()
+
+        if not unk_token:
```

Comment:
Are we expect users to pass `None` to the keyword argument `unk_token` here?

Reply:
We generally don't expect this. I was just being explicit here in case the user did pass in a `None` we would fail elegantly.